src/main/java/org/jenkinsci/plugins/workflow/libs/ResourceStep.java

```diff
@@ -74,10 +85,13 @@ public DescriptorImpl() {
 
         @Override protected String run() throws Exception {
             String resource = step.resource;
+            String libraryName = step.libraryName;
             Map<String,String> contents = LibraryAdder.findResources((CpsFlowExecution) getContext().get(FlowExecution.class), resource);
-            if (contents.isEmpty()) {
+            if (libraryName != null && contents.containsKey(libraryName)) {
+                return contents.get(libraryName);
+            } else if (contents.isEmpty() || (libraryName != null && !contents.containsKey(libraryName))) {
```

Comment:
I think it would be better to have different error message for the `libraryName != null && !contents.containsKey(libraryName))` condition saying the resource was found, but not in the specified library.